File: server/src/main/java/pt/isel/vsdserver/wss/WebSocketConfiguration.java
package pt.isel.vsdserver.wss;

import org.springframework.context.annotation.Configuration;
import org.springframework.messaging.simp.config.MessageBrokerRegistry;
import org.springframework.security.config.annotation.web.socket.AbstractSecurityWebSocketMessageBrokerConfigurer;
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
import org.springframework.web.socket.config.annotation.StompEndpointRegistry;

@Configuration
@EnableWebSocketMessageBroker
public class WebSocketConfiguration extends AbstractSecurityWebSocketMessageBrokerConfigurer {


    @Override
    public void configureMessageBroker(MessageBrokerRegistry config) {
        config.enableSimpleBroker("/topics");
        config.setApplicationDestinationPrefixes("/app");
    }

    @Override
    public void registerStompEndpoints(StompEndpointRegistry registry) {
        registry.addEndpoint("/my-ws")
                .setAllowedOrigins("*")
                .withSockJS();
    }


}

File: vsdashapi/src/main/kotlin/pt/isel/vsdashbapi/vsdashapi/communication/ws/WebSocketConfiguration.kt
package pt.isel.vsdashbapi.vsdashapi.communication.ws

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.messaging.simp.config.MessageBrokerRegistry
import org.springframework.web.socket.config.annotation.*
import pt.isel.vsdashbapi.vsdashapi.authentication.VsdUser.VsdSessionMapper

@Configuration
@EnableWebSocketMessageBroker
open class WebSocketConfiguration : WebSocketMessageBrokerConfigurer {


    override fun registerStompEndpoints(registry: StompEndpointRegistry){
        registry.addEndpoint("/app")
                .setHandshakeHandler(WebSocketHandshakeHandler())
                .withSockJS()                   //It appears to only work with sock js
                //.addInterceptors(HandshakeInterceptorImpl())
                //.setAllowedOrigins("*")
    }

    override fun configureMessageBroker(registry: MessageBrokerRegistry) {
        registry.enableSimpleBroker("/topic", "/queue")
        registry.setApplicationDestinationPrefixes("/api")
    }

}